src/new/sendTransaction.js

```diff
@@ -168,9 +169,67 @@ class SendTransaction extends EventEmitter {
       fullTxData.outputs = [...fullTxData.outputs, ...ret.data.outputs];
     }
 
+    this.fullTxData = fullTxData;
+    return fullTxData;
+  }
+
+  /**
+   * Prepare transaction data from inputs and outputs
+   * Fill the inputs if needed, create output change if needed and sign inputs
+   *
+   * @throws SendTxError
+   *
+   * @return {Transaction} Transaction object prepared to be mined
+   *
+   * @memberof SendTransaction
+   * @inner
+   */
+  prepareTx() {
+    if (!this.fullTxData) {
+      this.prepareTxData();
+    }
+    let preparedData = null;
+    try {
+      preparedData = transaction.prepareData(this.fullTxData, this.pin);
+      this.transaction = helpers.createTxFromData(preparedData, this.network);
+      return this.transaction;
+    } catch(e) {
+      const message = oldHelpers.handlePrepareDataError(e);
+      throw new SendTxError(message);
+    }
+  }
+
+  /**
+   * Prepare transaction data from inputs, outputs and signatures
+   * Fill the inputs if needed, create output change if needed
+   *
+   * @throws SendTxError
+   *
+   * @return {Transaction} Transaction object prepared to be mined
+   *
+   * @memberof SendTransaction
+   * @inner
+   */
+  prepareTxFrom(signatures) {
```

Comment:
I'm confused

The method is called `prepareTxFrom` and it receives `signatures`, but the docstring says it prepares the transaction from inputs, outputs and signatures

Should we update the docstring?

Reply:
Changed the docstring.

This method prepares the transaction to be mined from the signatures (and pre-prepared tx data).